cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -0,0 +1,232 @@
+
+	in := flag.String("in", "", "Input m3u8 manifest file")
+	concurrentSessions := flag.Int("concurrentSessions", 1, "# of concurrent transcode sessions")
+	segs := flag.Int("segs", 0, "Maximum # of segments to transcode (default all)")
+	transcodingOptions := flag.String("transcodingOptions", "P240p30fps16x9,P360p30fps16x9,P720p30fps16x9", "Transcoding options for broadcast job, or path to json config")
+	nvidia := flag.String("nvidia", "", "Comma-separated list of Nvidia GPU device IDs to use for transcoding")
+	outPrefix := flag.String("outPrefix", "", "Output segments' prefix (no segments are generated by default)")
+
+	flag.Parse()
+
+	if *in == "" {
+		glog.Errorf("Please provide the input manifest as `%s -in <input.m3u8>`", os.Args[0])
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	profiles := parseVideoProfiles(*transcodingOptions)
+
+	f, err := os.Open(*in)
+	if err != nil {
+		glog.Fatal("Couldn't open input manifest: ", err)
+	}
+	p, _, err := m3u8.DecodeFrom(bufio.NewReader(f), true)
+	if err != nil {
+		glog.Fatal("Couldn't decode input manifest: ", err)
+	}
+	pl, ok := p.(*m3u8.MediaPlaylist)
+	if !ok {
+		glog.Fatalf("Expecting media playlist in the input %s", *in)
+	}
+
+	accel := ffmpeg.Software
+	devices := []string{}
+	if *nvidia != "" {
+		accel = ffmpeg.Nvidia
+		devices = strings.Split(*nvidia, ",")
+	}
+
+	ffmpeg.InitFFmpeg()
+	var wg sync.WaitGroup
+	dir := path.Dir(*in)
+	start := time.Now()
+
+	table := tablewriter.NewWriter(os.Stderr)
+	data := [][]string{
+		{"Source File", *in},
+		{"Transcoding Options", *transcodingOptions},
+		{"Concurrent Sessions", fmt.Sprintf("%v", *concurrentSessions)},
+	}
+
+	for _, v := range data {
+		table.Append(v)
+	}
+
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetCenterSeparator("*")
+	table.SetColumnSeparator("|")
+	table.Render()
+
+	fmt.Println("timestamp,session,segment,transcode_time")
+	segCount := 0
+	segTotalDur := 0.0
+	for i := 0; i < *concurrentSessions; i++ {
+		wg.Add(1)
+		go func(k int, wg *sync.WaitGroup) {
+			tc := ffmpeg.NewTranscoder()
+			for j, v := range pl.Segments {
+				if *segs > 0 && j >= *segs {
+					break
+				}
+				if v == nil {
+					continue
+				}
+				u := path.Join(dir, v.URI)
+				in := &ffmpeg.TranscodeOptionsIn{
+					Fname: u,
+					Accel: accel,
+				}
+				if ffmpeg.Software != accel {
+					in.Device = devices[k%len(devices)]
+				}
+				profs2opts := func(profs []ffmpeg.VideoProfile) []ffmpeg.TranscodeOptions {
+					opts := []ffmpeg.TranscodeOptions{}
+					for n, p := range profs {
+						oname := ""
+						if *outPrefix != "" {
+							oname = fmt.Sprintf("%s_%s_%d_%d_%d.ts", *outPrefix, p.Name, n, k, j)
+						} else {
+							oname = "-"
+						}
+						o := ffmpeg.TranscodeOptions{
+							Oname:        oname,
+							Profile:      p,
+							Accel:        accel,
+							AudioEncoder: ffmpeg.ComponentOptions{Name: "drop"},
+							Muxer:        ffmpeg.ComponentOptions{Name: "mpegts"},
```

Comment:
We don't have have to explicitly set the null muxer here? Does the null muxer get used automatically if the Oname is set to "-"?

Reply:
Yepp exactly. I set it back to "null" here https://github.com/livepeer/go-livepeer/pull/1676/commits/7837b36a126b3057a65963de7c605adf32ad513c but that lead to `transcoder: Unable to open output` error for me on Nvidia

Reply:
Cool.

> lead to transcoder: Unable to open output error for me on Nvidia

Weird. I remember encountering something like that before, but IIRC the problem was that I did not enable the null muxer in the install_ffmpeg.sh script. 🤷‍♂️

Reply:
Also looks like `null` muxer doesn't need to be explicitly added in `./configure` script as well, so I will remove that line from `install_ffmpeg.sh` that I added here https://github.com/livepeer/go-livepeer/pull/1676/commits/fce6bcf8b72e47f507a34bbad3305eb13218dc31

Reply:
>  I remember encountering something like that before, but IIRC the problem was that I did not enable the null muxer in the install_ffmpeg.sh script.

Oh wait you are right, I forgot to run `install_ffmpeg.sh` on my GPU machine.

I'll revert it back to the status-quo of having "null" for "-" and "mpegts" for "prefix/output.ts"